src/Tasks/Touch.cs

```diff
@@ -185,9 +226,12 @@ public override bool Execute()
                     return false;
                 }
             }
-            else
+
+            Log.LogMessageFromResources(messageImportance, "Touch.Touching", file);
+
+            if (FailIfNotIncremental)
             {
-                Log.LogMessageFromResources(MessageImportance.Normal, "Touch.Touching", file);
+                return true;
```

Comment:
I don't understand this. Isn't `Touch` inherently non-incremental since it'll update the modtime? Shouldn't this be a failure case?

Reply:
Touch function itself is non-incremental, but in practice, the touch task is like an Exec task for "touch.exe". It is wrapped with incremental output check such that as long as the output exists, then it can skip.

Another way to think about is how to check if this task ran last build.

Reply:
Sorry, this isn't clicking with me. In a non-question build, if we touch an input to any later target/task, it'll break incremental behavior downstream. Why wouldn't question report that?

Reply:
Good catch.

Reply:
After some testing, I have found this case in microsoft.net.compilers.toolset \ Microsoft.Managed.Core.targets

  <!-- We touch and create a semaphore file after build to indicate the time stamp for last build with skipAnalyzers flag. -->
  <Target Name="_TouchLastBuildWithSkipAnalyzers" Condition="'$(_SkipAnalyzers)' == 'true'" AfterTargets="CoreCompile">
    <Touch AlwaysCreate="true" Files="$(_LastBuildWithSkipAnalyzers)"/>
  </Target>

@rainersigwald , Do you have thoughts about how strict should question mode be.  If the touch task doesn't affect the build, should it error?

Reply:
How are you determining `doesn't affect the build` at the time when you touched it?

Reply:
> 

This file doesn't affect the inputs and outputs of build.  For example, some tools will touch a file to trigger a file watcher to start testing or analysis.

Reply:
> 

No way determining the effect.  It is all or none.

Reply:
So in that case, it sounds like Touch should just never be "incremental" to me.

Reply:
Sounds good.